it/src/test/scala/com/wavesplatform/it/sync/smartcontract/smartasset/NoOrderProofsSuite.scala

```diff
@@ -27,18 +29,17 @@ class NoOrderProofsSuite extends BaseTransactionSuite {
         script = Some(
           ScriptCompiler(
             s"""
-            match tx {
-            case o: Order => true
-            case _ => false
-            }""".stripMargin,
-            true
+              |match tx {
+              |  case o: Order => true
+              |  case _ => false
+              |}""".stripMargin,
+            isAssetScript = true
           ).explicitGet()._1.bytes.value.base64)
       )
-    catch {
-      case ex: java.lang.Exception => assert(ex.getMessage.contains("Compilation failed: Matching not exhaustive"))
-      case _: Throwable            => throw new Exception("ScriptCompiler works incorrect for orders with smart assets")
+    } match {
+      case Success(_) => fail("ScriptCompiler didn't throw expected error")
```

Comment:
I don't think that we should here use "some sort of sugar" for "better reading". My point was to catch exactly java.lang.Exception exception, if we will customise our exception this test catches that. But in your interpretation we won't see changes.

Reply:
Previous check was not exact catching too. It catched any exception inherited of java.lang.Exception.